setoff_workflows/setoff_workflows.py

```diff
@@ -0,0 +1,2622 @@
+
+            # Get run-wide commands that apply to all sequencing runs
+            dx_cmd_list.extend(self.return_multiqc_cmds())
+
+            if self.samples_obj.pipeline == "pipe":
+                # We want duty_csv to also depend on the cnv calling jobs for PIPE workflows
+                dx_cmd_list.append(SWConfig.UPLOAD_ARGS["depends_list_cnv_recombined"])
+
+            dx_cmd_list.append(self.create_duty_csv_cmd())
+
+            self.rf_obj.rf_loggers.sw.info(
+                self.rf_obj.rf_loggers.sw.log_msgs["cmds_built"]
+            )
+        return dx_cmd_list, dx_postprocessing_cmds, decision_support_upload_cmds
+
+    def return_tso_runwide_cmds(self) -> Union[list, list, list]:
+        """
+        Collect runwide commands for tso500 runs as a list. This includes tso500 app,
+        fastqc, sompy, sambamba, multiqc and duty_csv. TSO commands are all generated
+        within this function as the dependency order is different for this pipeline
+            :return dx_cmd_list (list):                     List of runwide commands for tso runs
+            :return dx_postprocessing_cmds (list):          Post-processing commands for tso runs
+            :return decision_support_upload_cmds (list):    Commands for decision support uploads
+        """
+        dx_cmd_list = [
+            f"PROJECT_ID={self.nexus_project_id}",
+            f"PROJECT_NAME={self.samples_obj.nexus_paths['proj_name']}",
+            f"RUNFOLDER_NAME={self.rf_obj.runfolder_name}",
+            SWConfig.EMPTY_DEPENDS,
+        ]
+        dx_postprocessing_cmds = dx_cmd_list[:]
+        decision_support_upload_cmds = dx_cmd_list[:]
+        sambamba_cmds_list = []
+        # Remove base SampleSheet as we only want to use split SampleSheets
+        for tso_ss in self.rf_obj.tso_ss_list:
+            if tso_ss != self.rf_obj.samplesheet_name:
+                dx_cmd_list.append(self.create_tso500_cmd(tso_ss))
+
+        for sample_name in self.samples_obj.samples_dict.keys():
+            decision_support_upload_cmds.append(
+                self.samples_obj.samples_dict[sample_name][
+                    "decision_support_upload_cmd"
+                ]
+            )
+            # Append all fastqc commands to cmd_list
+            dx_postprocessing_cmds.append(
+                self.samples_obj.samples_dict[sample_name]["sample_pipeline_cmd"]
+            )
+            dx_postprocessing_cmds.append(SWConfig.UPLOAD_ARGS["depends_list"])
+            sambamba_cmds_list.append(
+                self.create_sambamba_cmd(
+                    sample_name, self.samples_obj.samples_dict[sample_name]["pannum"]
+                )
+            )
+            # Exclude negative controls from the depends list as the NTC coverage
+            # calculation can often fail. We want the coverage report for the NTC sample
+            # to help assess contamination. Only add to depends_list if job ID from
```

Comment:
It is only adding the command that adds the job ID to the dependency list for samples that are not negative controls which would be the desired behaviour, as if the negative control coverage job fails we do not want this to prevent downstream apps from running